_posts/en/newsletters/2025-03-14-newsletter.md

```diff
@@ -0,0 +1,222 @@
+    on the top of the stack is `0`, they use the `HASH160` opcode;
+    when the value is `1`, they use the `SHA256` opcode.  In Alice's
+    case, she performs `sha256(hash160(hash160(alice_nonce)))`; in Bob's
+    case, he performs `sha256(sha256(hash160(bob_none)))`.  This
+    produces a commitment for each of them, which they send to each
+    other without revealing either their value or their nonce.
+
+  - With the commitments shared, they create an onchain funding
+    transaction with a script that will validate the inputs using
+    `OP_IF` to choose between the different hash functions and allows
+    one of them to claim the payment.  For example, if the sum of their
+    two xored values is 0 or 1, Alice receives the money; if it's 2 or
+    3, Bob receives it.  The contract may also have a timeout clause and
+    a space-saving mutual-agreement clause.
+
+  - After the funding transaction confirms to a suitable depth, Alice
+    and Bob disclose to each other their values and their nonces.  The
+    xor of `1 0 0` and `1 1 0` is `0 1 0`, which sums to `1`, allowing
+    Alice to claim the payment.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review
+Club][] meeting, highlighting some of the important questions and
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[Stricter internal handling of invalid blocks][review club 31405] is a
+PR by [mzumsande][gh mzumsande] that improves the correctness of two
+non-consensus-critical and expensive-to-calculate validation fields by
+immediately updating them when a block is marked as invalid. Prior to
+this PR, these updates were delayed to a later event to minimize
+resource usage. However, since [Bitcoin Core #25717][] an attacker would
+need to invest much more work to exploit this.
+
+Specifically, this PR ensure that `ChainstateManager`'s `m_best_header`
+always points to the most-work header not known to be valid, and that
+a block's `BLOCK_FAILED_CHILD` `nStatus` is always correct.
+
+
+{% include functions/details-list.md
+  q0="Which purpose(s) does `ChainstateManager::m_best_header` serve?"
+  a0="`m_best_header` represents the most-PoW header the node has soon
+  so far which it hasn't yet invalidated but also can't guarantee to be
+  valid. It has many uses, but the main one is to serve as a target to
+  which the node can progress its best chain. Other use cases include
+  providing an estimate of the current time, and an estimate of the best
+  chain height when requesting missing headers from a peer. A more
+  complete overview can be found in the ~6-year old pull request
+  [Bitcoin Core #16974][]."
+  a0link="https://bitcoincore.reviews/31405#l-36"
+
+  q1="Prior to this PR, which of these statements are true, if any? 1) a
+	`CBlockIndex` with an INVALID predecessor will ALWAYS have a
+	`BLOCK_FAILED_CHILD` `nStatus`. 2) a `CBlockIndex` with an VALID
```

Comment:
```suggestion
	`BLOCK_FAILED_CHILD` `nStatus`. 2) a `CBlockIndex` with a VALID
```
(also, should VALID be code-formatted?)

Reply:
Thanks, applied the suggested fix. I don't think VALID should be code formatted, it is not a status or enum member. It is capitalized to highlight the difference with INVALID in option 1 (similar to ALWAYS/NEVER).